shared/src/main/scala/mlscript/JSBackend.scala

```diff
@@ -518,8 +518,11 @@ class JSTestBackend extends JSBackend {
           val translatedBody = translateTerm(body)(topLevelScope)
           val jsName = topLevelScope declare mlsName
           S(
-            S(JSLetDecl.from(jsName :: topLevelScope.getTempVars())) ->
-             ((JSIdent(jsName) := translatedBody) ::
+            topLevelScope.emitTempVarDecls() ->
+             ((JSIdent("globalThis").member(jsName) := (translatedBody match {
```

Comment:
By the way, why `globalThis`? It may contain other definitions that will clash with the ones we register. All we need is a simple local dictionary.

Reply:
Because I want to make sure that if some errors occur during the initialization of a variable, it will not declare the name. By doing so, future expressions which reference the variable can throw `ReferenceError` rather than evaluate `undefined` and produce some strange values.

For example, with the previous execution strategy, it produces `NaN` for the second test block.

```
def a = <something error>
// Generated code:
// let a;
// try { a = <...>; res = a; } catch (e) { ... }
// Runtime error: ...

a + 1
// Result: NaN
```

If we use the current approach, the reference to `a` causes `ReferenceError`. 

```
def a = <something error>
// Generated code:
// try { someResultDictionary.a = <...>; res = a; } catch (e) { ... }
// Runtime error: ...

a + 1
// Runtime error: ReferenceError: a is not defined
```

The reason why I used `globalThis` are as follows.

1. in Node.js REPL, declare a un-scoped variable is equivalent to writing it to the global object. Here is an example from REPL.
   
   ```
   ❯ node
   Welcome to Node.js v17.2.0.
   Type ".help" for more information.
   > a = 0
   0
   > globalThis.a
   0
   > var b = 5
   undefined
   > globalThis.b
   5
   ```
2. I have to replace all code generation for identifiers by member access of that local dictionaries.

Reply:
Ah, ok sure.

Reply:
But I agree with you, `globalThis` may cause conflicts. So I'm implementing the 2 in my previous commit.

Reply:
I’ll leave this as future work.